internal/devices/create.go

```diff
@@ -64,62 +66,125 @@ func (c *Client) Create() *cobra.Command {
 			}
 			cmd.SilenceUsage = true
-
 			if userdataFile != "" {
 				userdataRaw, readErr := os.ReadFile(userdataFile)
 				if readErr != nil {
-					return fmt.Errorf("Could not read userdata-file: %w", readErr)
+					return fmt.Errorf("could not read userdata-file: %w", readErr)
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
-					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
+					return fmt.Errorf("could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
+			}
+			var publicSubnet int32
+			// In case of no configuration,
+			if publicIPv4SubnetSize == 0 {
+				publicSubnet = 31 // Assigin default to 31
+			} else {
+				publicSubnet = int32(publicIPv4SubnetSize)
 			}
 
+			formatedCdata := make(map[string]interface{})
+			if customdata != "" {
+				err := json.Unmarshal([]byte(customdata), &formatedCdata)
+				if err != nil {
+					return fmt.Errorf("could not convert customedata :%w", err)
+				}
+			}
 			var facilityArgs []string
+
+			var request metal.ApiCreateDeviceRequest
 			if facility != "" {
 				facilityArgs = append(facilityArgs, facility)
-			}
 
-			request := &packngo.DeviceCreateRequest{
-				Hostname:              hostname,
-				Plan:                  plan,
-				Facility:              facilityArgs,
-				Metro:                 metro,
-				OS:                    operatingSystem,
-				BillingCycle:          billingCycle,
-				ProjectID:             projectID,
-				UserData:              userdata,
-				CustomData:            customdata,
-				IPXEScriptURL:         ipxescripturl,
-				Tags:                  tags,
-				PublicIPv4SubnetSize:  publicIPv4SubnetSize,
-				AlwaysPXE:             alwaysPXE,
-				HardwareReservationID: hardwareReservationID,
-				SpotInstance:          spotInstance,
-				SpotPriceMax:          spotPriceMax,
-				TerminationTime:       endDt,
+				facilityDeviceRequest := metal.CreateDeviceRequest{
+					DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+						Facility:        facilityArgs,
+						Plan:            plan,
+						OperatingSystem: operatingSystem,
+						Hostname:        &hostname,
+						Userdata:        &userdata,
+					},
+				}
+
+				if billingCycle != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetBillingCycle(billingCycle)
+				}
+
+				if alwaysPXE {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetAlwaysPxe(alwaysPXE)
+				}
+
+				if ipxescripturl != "" {
+					facilityDeviceRequest.DeviceCreateInFacilityInput.SetIpxeScriptUrl(ipxescripturl)
+				}
+
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetPublicIpv4SubnetSize(publicSubnet)
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetTerminationTime(endDt)
+				facilityDeviceRequest.DeviceCreateInFacilityInput.SetTags(tags)
```

Comment:
It should be safe to set `Tags: tags` in the initial definition of `facilityDeviceRequest` (same for `metroDeviceRequest`), since the tags property is an array of strings.